### PR TITLE
feat: missing api features

### PR DESCRIPTION
**Pull request by sofushn**

### CODE REVIEW

File: src/CitMovie.Business/Managers/PersonManager.cs [hunk 1]
```diff
@@ -12,17 +12,24 @@ public PersonManager(IPersonRepository personRepository, IMapper mapper)
     }
 
 
-    public async Task<IEnumerable<PersonResult>> GetPersonsAsync(int page, int pageSize)
-
+    public async Task<IEnumerable<PersonResult>> QueryPersonsAsync(PersonQueryParameter queryParameter, PageQueryParameter pageQuery, int? userId)
```
Comment (JuliusKryger) [inline src/CitMovie.Business/Managers/PersonManager.cs:15]: Maybe we should have some null check for the queryParameter. The QueryPerson endpoint does not check if queryParameter is null before using it. I feel like we should make this check one of the places

File: src/CitMovie.Data/Repositories/PersonRepository.cs [hunk 2]
```diff
@@ -17,43 +17,136 @@ public async Task<IEnumerable<Person>> GetPersonsAsync(int page, int pageSize)
             .Pagination(page, pageSize)
             .ToListAsync();
     }
-    public async Task<int> GetTotalPersonsCountAsync()
-    {
-        return await _dataContext.People.CountAsync();
+
+    public async Task<IEnumerable<Person>> GetPersonsByNameAsync(string name, int? userId, int page, int pageSize) {
+        IQueryable<int> ids = _dataContext
+            .PersonSearch(name, userId)
```
Comment (JuliusKryger) [inline src/CitMovie.Data/Repositories/PersonRepository.cs:23]: Now when we are using our own SQL function "structured_string_search_name" are we then sure that it is safe against SQL-Injection? Here in this function we are just passing the raw data to the DB function, so we have no protection here in the backend.

**Review by JuliusKryger — CHANGES_REQUESTED** [review]

File: src/CitMovie.Business/Managers/PersonManager.cs (hunk 1 above)
Comment (sofushn) [reply src/CitMovie.Business/Managers/PersonManager.cs:15]: This is not a problem, the aspnet framework initializes a new instance of the objects regardless, if you look at the properties in the `PersonQueryParameter ` and `PageQueryParameter ` they have been specified as nullable or not.

File: src/CitMovie.Data/Repositories/PersonRepository.cs (hunk 2 above)
Comment (sofushn) [reply src/CitMovie.Data/Repositories/PersonRepository.cs:23]: Entitiy framework uses parameters for all generated queries. See example on picture

image

**Pull request merged**